lib/os/mem_blocks.c

```diff
@@ -16,37 +16,40 @@ static void *alloc_blocks(sys_mem_blocks_t *mem_block, size_t num_blocks)
 	size_t offset;
 	int r;
 	uint8_t *blk;
-	void *ret = NULL;
 
 #ifdef CONFIG_SYS_MEM_BLOCKS_RUNTIME_STATS
 	k_spinlock_key_t  key = k_spin_lock(&mem_block->lock);
 #endif
 
 	/* Find an unallocated block */
-	r = sys_bitarray_alloc(mem_block->bitmap, num_blocks, &offset);
-	if (r == 0) {
-
+	r = sys_bitarray_alloc(mem_block->bitmap, info.num_blocks, &offset);
```

Comment:
This `info.num_blocks` change shouldn't be here - it is fixed down the line in another patch, but things would be simpler without this extraneous change.

Reply:
Fixing that ...